src/Dumplie/Domain/Customer/CartItem.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+declare (strict_types = 1);
+
+namespace Dumplie\Domain\Customer;
+
+use Dumplie\Domain\Customer\Exception\InvalidArgumentException;
+use Dumplie\Domain\SharedKernel\Money\Price;
+
+final class CartItem
+{
+    /**
+     * @var Product
+     */
+    private $product;
```

Comment:
There is no need to store whole `Product` in `CartItem` - SKU is enough.
For all calculations fresh Product entities should be used.
Product prices should be frozen only after placing an order.

Therefore static constructor would be handy:
```php
CartItem::fromProduct(Product $product, int $quantity);
CartItem::fromSKU(SKU $sku, int $quantity);
```